cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -52,12 +67,15 @@
   private final LocationFactory locationFactory;
   private final HBaseTableUtil hBaseTableUtil;
   private final DatasetFramework dsFramework;
+  private final TransactionExecutorFactory transactionExecutorFactory;
   private final Pattern userTablePrefix;
 
   @Inject
-  private DatasetUpgrader(CConfiguration cConf, Configuration hConf, LocationFactory locationFactory,
-                          NamespacedLocationFactory namespacedLocationFactory,
-                          HBaseTableUtil hBaseTableUtil, DatasetFramework dsFramework) {
+  @VisibleForTesting
+  DatasetUpgrader(CConfiguration cConf, Configuration hConf, LocationFactory locationFactory,
```

Comment:
can be private and remove VisibleForTesting. We are not using this in any tests yet.